src/service/search/sql.rs

```diff
@@ -591,18 +591,12 @@ impl Sql {
         true
     }
 
-    /// filter source by partition key
-    pub async fn filter_source_by_partition_key(&self, source: &str) -> bool {
-        // check partition key
-        for key in &self.meta.quick_text {
-            let field = logs::get_partition_key_query(format!("{}=", key.0).as_str());
-            let value = logs::get_partition_key_query(format!("{}={}", key.0, key.1).as_str());
-            if str::find(source, &format!("/{field}")) && !str::find(source, &format!("/{value}/"))
-            {
-                return false;
-            }
-        }
-        true
+    pub(crate) fn filter_source_by_partition_key(&self, source: &str) -> bool {
+        !self.meta.quick_text.iter().any(|(k, v, _)| {
+            let field = logs::get_partition_key_query(&format!("{k}="));
+            let value = logs::get_partition_key_query(&format!("{k}={v}"));
+            str::find(source, &format!("/{field}")) && str::find(source, &format!("/{value}/"))
+        })
```

Comment:
Drat! I've fat-fingered the exclamation mark before the second `str::find` away! 😳

Great job spotting this, Hengfei!

I've fixed the error, renamed `str::find` to `find` (to avoid confusion with [the library function](https://doc.rust-lang.org/std/primitive.str.html#method.find)), and added a unit test.

Thanks!